cpp/include/cudf/groupby.hpp

```diff
@@ -65,6 +65,20 @@ struct aggregation_result {
   std::vector<std::unique_ptr<column>> results{};
 };
 
+/**
+ * @brief The grouped data corresponding to a groupby operation on a set of values.
+ *
+ * A `groupby_groups` object holds two tables of identical number of rows:
+ * a table of grouped keys and a table of grouped values. In addition, it holds
+ * a vector of integer offsets into the rows of the tables, such that
+ * `group_offsets[i+1] - group_offsets[i]` gives the size of group `i`.
+ */
+struct groupby_groups {
```

Comment:
If we move this struct definition to inside the `groupby` object, that eliminates the need for the prefix.

E.g.,

```
class groupby{
   ...
   struct groups{
      ...
   };
};

Reply:
I did that, renaming the `groups()` function to `get_groups()`.